File: TCDNew-master/src/ui.android/lib/services/GStateProvider.dart
import 'package:tiq/models/state/SettingsApp.dart';
import 'package:tiq/models/state/SettingsFacility.dart';
import 'package:tiq/models/state/SettingsUser.dart';
import 'package:tiq/models/state/StateAuth.dart';

class GStateProvider {
  GStateProvider._internal();
  static final GStateProvider _singleton = new GStateProvider._internal();
  static GStateProvider get instance => _singleton;

  SettingsApp _settingsApp;
  SettingsUser _settingsUser;
  SettingsFacility _settingsFacility;
  StateAuth _stateAuth;

  SettingsApp get settingsApp => _settingsApp;
  SettingsUser get settingsUser => _settingsUser;
  SettingsFacility get settingsFacility => _settingsFacility;
  StateAuth get stateAuth => _stateAuth;

  void setSettingsUser(SettingsUser settingsUser) {
    this._settingsUser = settingsUser;
  }

  void setSettingsApp(SettingsApp settingsApp) {
    this._settingsApp = settingsApp;
  }

  void setSettingsFacility(SettingsFacility settingsFacility) {
    this._settingsFacility = settingsFacility;
  }

  void setStateAuth(StateAuth stateAuth) {
    this._stateAuth = stateAuth;
  }
}

File: TCDNew-master/src/ui.wince/Cen.Wms.Client/Services/GStateProvider.cs
using Cen.Wms.Client.Models.State;

namespace Cen.Wms.Client.Services
{
    class GStateProvider
    {
        private static GStateProvider _instance;

        public static GStateProvider Instance
        {
            get
            {
                if (_instance == null)
                    _instance = new GStateProvider();

                return _instance;
            }
        }

        private SettingsApp _settingsApp;
        private SettingsUser _settingsUser;
        private SettingsFacility _settingsFacility;
        private StateAuth _stateAuth;

        public SettingsApp SettingsApp { get { return _settingsApp; } }
        public SettingsUser SettingsUser { get { return _settingsUser; } }
        public SettingsFacility SettingsFacility { get { return _settingsFacility; } }
        public StateAuth StateAuth { get { return _stateAuth; } }

        public void SetSettingsUser(SettingsUser settingsUser)
        {
            this._settingsUser = settingsUser;
        }

        public void SetSettingsApp(SettingsApp settingsApp)
        {
            this._settingsApp = settingsApp;
        }

        public void SetSettingsFacility(SettingsFacility settingsFacility)
        {
            this._settingsFacility = settingsFacility;
        }

        public void SetStateAuth(StateAuth stateAuth)
        {
            this._stateAuth = stateAuth;
        }
    }
}
